### DISCUSSION TITLE
Refreshing JWT Session in api endpoint?

### DISCUSSION THREAD

**Discussion by KoesterJannik** [nextauthjs/next-auth#4881] [Help]
`// Update payment to PAID status once completed
  await prisma.payment.updateMany({
    where: {
      orderID,
    },
    data: {
      status: "PAID",
    },
  });
  await prisma.user.update({
    where: {
      id: user.id,
    },
    data: {
      role: "PREMIUM",
    },
  });` Right now the user has to log out and in before the  role in the session has changed, is there a way to reset the next-auth cookie and refresh the session?

**Comment by ThangHuuVu** [comment]
You could call the session route and the session will be updated, either on the client side via `getSession` or in the API route using `unstable_getServerSession`. Let me know if this helps 👍

**Reply by KoesterJannik** [reply]
I tried your first attempt but i didnt worked.
What i did.
1. User is at the site to buy premium.
```const captureMutation = useMutation<string, AxiosError, any, Response>(
    
    async (data): Promise<void> => {
      await axios.post("/api/paypal/captureOrder", data);
      // At this point the user has a premium role, so redirecting them to their dashboard, where useSession hook is called.
      await router.push("/me");
    }
  );
```


At this point he has the role PREMIUM in the database, as you said i redirected him to their dashboard where i call useSession like so:

```
import { useSession } from "next-auth/react";

import IsAuthenticated from "../../components/wrapper/IsAuthenticated";

export default function MePage() {
  const { data } = useSession();

  return (
    <IsAuthenticated>
      <pre>{JSON.stringify(data, null, 2)}</pre>
    </IsAuthenticated>
  );
}
```

But the session still hase the normal "USER" Role instead of the premium. Right now a relogin needs to be done. Am i missing something?

Thanks btw, really appreciate your help ! :)
